File: ai-engine/app/routers/contracts/game_interface.py
from enum import Enum
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class GameStatus(str, Enum):
    active = "active"
    stopped = "stopped"


class Actor(str, Enum):
    user = "user"
    agent = "agent"


class GameMove(BaseModel):
    actor: Actor
    order: int
    notation: str
    message: str = ""


class GameInterface(BaseModel):
    id: str
    userName: str
    engineLevel: str
    moves: list[GameMove] = []
    status: GameStatus
    startedAt: Optional[datetime] = None


class MoveRequest(BaseModel):
    game_uuid: str
    actor: str
    order: int
    notation: str

File: backend/src/game/contracts/game.interface.ts
export enum GameStatus {
  active = 'active',
  stopped = 'stopped',
}

export enum Actor {
  user = 'user',
  agent = 'agent',
}

export type EngineLevel = 'Amateur' | 'Intermediate' | 'Professional';

export interface GameMove {
  actor: Actor;
  order: number;
  notation: string;
  message: string;
}

export interface GameInterface {
  id: string;
  userName: string;
  engineLevel: EngineLevel;
  moves: GameMove[];
  status: GameStatus;
  startedAt: Date;
}

export interface GameHistoryItem {
  id: string;
  userName: string;
  engineLevel: EngineLevel;
  startedAt: Date;
  totalMoves: number;
}
